src/main/java/carpet/script/language/ControlFlow.java

```diff
@@ -42,26 +44,80 @@ public static void apply(Expression expression) // public just to get the javado
 
         expression.addFunction("exit", (lv) -> { throw new ExitStatement(lv.size()==0?Value.NULL:lv.get(0)); });
 
-        expression.addFunction("throw", (lv)-> {throw new ThrowStatement(lv.size()==0?Value.NULL:lv.get(0)); });
+        expression.addLazyFunction("throw", -1, (c, t, lv)-> 
+        {
+            switch (lv.size()) 
+            {
+                case 0:
+                    c.host.issueDeprecation("throw() (without id)");
+                    throw new ThrowStatement(new StringValue("No further information"), Value.NULL, Value.NULL);
+                case 1:
+                    throw new ThrowStatement(lv.get(0).evalValue(c));
+                case 2:
+                    throw new ThrowStatement(lv.get(1).evalValue(c), lv.get(0).evalValue(c), Value.NULL);
+                case 3:
+                    throw new ThrowStatement(lv.get(1).evalValue(c), lv.get(0).evalValue(c), lv.get(2).evalValue(c));
+                default:
+                    throw new InternalExpressionException("throw() can't accept more than 3 parameters");
+            }
+        });
 
         expression.addLazyFunction("try", -1, (c, t, lv) ->
         {
             if (lv.size()==0)
-                throw new InternalExpressionException("'try' needs at least an expression block");
+                throw new InternalExpressionException("'try' needs at least an expression block, a filter, and a catch_epr");
             try
             {
                 Value retval = lv.get(0).evalValue(c, t);
                 return (c_, t_) -> retval;
             }
-            catch (ThrowStatement ret)
+            catch (ProcessedThrowStatement ret)
             {
+                if (ret.exception.isError())
+                    throw ret;
                 if (lv.size() == 1)
+                {
+                    c.host.issueDeprecation("try(expr) (without filter and catch expression)");
                     return (c_, t_) -> Value.NULL;
+                }
+                // This is outside since it is used in try(expr,catch_expr)
                 LazyValue __ = c.getVariable("_");
-                c.setVariable("_", (__c, __t) -> ret.retval.reboundedTo("_"));
-                Value val = lv.get(1).evalValue(c, t);
+                c.setVariable("_", (__c, __t) -> ret.exception.getValue().reboundedTo("_"));
+                LazyValue __msg = c.getVariable("_msg");
+                c.setVariable("_msg", (__c, __t) -> StringValue.of(ret.message).reboundedTo("_msg"));
```

Comment:
yeah - with value cointaining the information about the exception, there is no need for message (_msg) - the value thrown is the message, which could be anything.